### ISSUE TITLE
Basic UI HTTP ERROR 500 java.lang.NullPointerException

### ISSUE THREAD

**Issue by grzegorz914** [openhab/openhab-webui#1707]
## Which UI are you reporting an issue for?

- [x] Basic UI
- [ ] HABPanel
- [ ] HABot
- [ ] CometVisu

## The problem

ERROR 500 java.lang.NullPointerException

## Expected behavior

Correct load.

## Steps to reproduce

install latest snapshot 3313

## Your environment

```
OH 4.0.0. 3313
```

## Browser console

```
HTTP ERROR 500 java.lang.NullPointerException: Cannot invoke "java.util.Map.put(Object, Object)" because "this.global" is null

Caused by:

java.lang.NullPointerException: Cannot invoke "java.util.Map.put(Object, Object)" because "this.global" is null
	at com.oracle.truffle.js.scriptengine.GraalJSBindings.put(GraalJSBindings.java:130)
	at java.scripting/javax.script.SimpleScriptContext.setAttribute(SimpleScriptContext.java:246)
	at org.openhab.core.automation.module.script.ScriptTransformationService.transform(ScriptTransformationService.java:145)
	at org.openhab.core.ui.internal.items.ItemUIRegistryImpl.transform(ItemUIRegistryImpl.java:554)
	at org.openhab.core.ui.internal.items.ItemUIRegistryImpl.getLabel(ItemUIRegistryImpl.java:462)
	at org.openhab.ui.basic.internal.render.AbstractWidgetRenderer.preprocessSnippet(AbstractWidgetRenderer.java:101)
	at org.openhab.ui.basic.internal.render.TextRenderer.renderWidget(TextRenderer.java:57)
	at org.openhab.ui.basic.internal.render.PageRenderer.renderWidget(PageRenderer.java:197)
	at org.openhab.ui.basic.internal.render.PageRenderer.processChildren(PageRenderer.java:163)
	at org.openhab.ui.basic.internal.render.PageRenderer.processChildren(PageRenderer.java:184)
	at org.openhab.ui.basic.internal.render.PageRenderer.processPage(PageRenderer.java:127)
	at org.openhab.ui.basic.internal.servlet.WebAppServlet.service(WebAppServlet.java:182)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:550)
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:74)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440)
	at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:294)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:90)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.base/java.lang.Thread.run(Thread.java:833)

```

**Comment by lolodomo** [comment]
@J-N-K : do you think there is a link with the other core issues discovered and/or fixed today?

**Comment by grzegorz914** [comment]
build 3316 the issue still exist

**Comment by J-N-K** [comment]
At least it's not an UI issue. I have seen these NPEs also while working on https://github.com/openhab/openhab-addons/pull/14398, but never before or after that. For me the NPE is thrown every second or third time I reload a rule from file (or save it from UI) without that fix. I don't fully understand why `globals` can be null, but maybe the underlying issue is that something goes wrong in the script engine when initializing context fails due to an error. IF this is the case, then I would say it is an external bug in GraalVM, but can't say for sure.

TL;DR: We can probably ignore it after the mentioned PR has been merged.

**Comment by grzegorz914** [comment]
build 3320 fixed the issue
